### PR TITLE
Homematic: Some updates

### PR DESCRIPTION
**Pull request by gerrieg**
Signed-off-by: Gerhard Riegler <gerhard.riegler@gmail.com>

### CODE REVIEW

**Comment by gerrieg** [comment]
The StateContact virtual datapoint has just been confirmed in #1564
I think it's ready to merge

File: addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicBridgeHandler.java [hunk 1]
```diff
@@ -238,7 +238,7 @@ public void onStateUpdated(HmDatapoint dp) {
         Thing hmThing = getThingByUID(UidUtils.generateThingUID(dp.getChannel().getDevice(), getThing()));
         if (hmThing != null && hmThing.getHandler() != null) {
             final ThingStatus status = hmThing.getStatus();
-            if (status == ThingStatus.ONLINE || status == ThingStatus.OFFLINE) {
+            if (status != null && (status == ThingStatus.ONLINE || status == ThingStatus.OFFLINE)) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicBridgeHandler.java]: null-check not needed those others checks are null safe (or do I miss something?)

File: addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/StateContactVirtualDatapointHandler.java [hunk 2]
```diff
@@ -71,12 +71,13 @@ private boolean isApplicable(HmDevice device) {
     }
 
     private Boolean convertState(Object value) {
-        if (value == null) {
+        if (value == null || !(value instanceof Integer)) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/StateContactVirtualDatapointHandler.java]: !(null instanceof Integer) => imho you can remove the null-check

File: addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/StateContactVirtualDatapointHandler.java [hunk 3]
```diff
@@ -71,12 +71,13 @@ private boolean isApplicable(HmDevice device) {
...
             return null;
         }
-        if ("CLOSED".equals(value)) {
+        int intValue = (int) value;
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/StateContactVirtualDatapointHandler.java]: This conversion is not needed, you checked it to be an instance of Integer and https://stackoverflow.com/a/30454826/3656774

File: addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicBridgeHandler.java (hunk 1 above)
Comment (gerrieg) [inline addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicBridgeHandler.java]: my fault, i saw a NPE but i had the wrong source code. The NPE has already been fixed in a previous PR

File: addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/StateContactVirtualDatapointHandler.java (hunk 3 above)
Comment (gerrieg) [inline addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/StateContactVirtualDatapointHandler.java]: OK, but then i have to cast it twice...

**Comment by martinvw** [comment]
thanks!

File: addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/StateContactVirtualDatapointHandler.java (hunk 3 above)
Comment (martinvw) [inline addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/StateContactVirtualDatapointHandler.java]: My point or assumption was that even that is not needed, does your ide or compiler force you to do so?

Comment (gerrieg) [inline addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/virtual/StateContactVirtualDatapointHandler.java]: yes, eclipse wants this cast

**Pull request merged**